File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], costs: List[int]) -> int:
      
        right_index = 0
        possible_count = 0
        reseted_count = 0
        for i in range(len(gas)):
            possible_count += gas[i]-costs[i]
            reseted_count += gas[i]-costs[i]
            if reseted_count < 0:
                reseted_count = 0
                right_index = i + 1
                
            if possible_count >= 0 and i == len(gas)-1:
                return right_index 
        return -1
              
            
            

File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int poss = 0;
        int poss_temp = 0;
        int idx  = 0;
        vector<int> d(gas.size() , 0);
        for(int i = 0; i < gas.size(); i++){
            int curr = gas[i] - cost[i];
            poss+=curr;
            poss_temp += curr;
            if(poss_temp < 0){
                poss_temp = 0;
                idx = i+1;
                
                
            }
            if(poss >= 0 && i == gas.size()-1){
                return idx;
            }
        }
        return -1;
        
    }
};